pkg/api/v1/types.go

```diff
@@ -181,6 +181,19 @@ type VirtualMachineInstanceSpec struct {
 	Subdomain string `json:"subdomain,omitempty"`
 	// List of networks that can be attached to a vm's virtual interface.
 	Networks []Network `json:"networks,omitempty"`
+	// Set DNS policy for the pod.
+	// Defaults to "ClusterFirst".
+	// Valid values are 'ClusterFirstWithHostNet', 'ClusterFirst', 'Default' or 'None'.
+	// DNS parameters given in DNSConfig will be merged with the policy selected with DNSPolicy.
+	// To have DNS options set along with hostNetwork, you have to specify DNS policy
+	// explicitly to 'ClusterFirstWithHostNet'.
+	// +optional
+	DNSPolicy k8sv1.DNSPolicy `json:"dnsPolicy,omitempty" protobuf:"bytes,6,opt,name=dnsPolicy,casttype=DNSPolicy"`
```

Comment:
Can you check if some wrong configurations on a pod are rejected by the cluster, when the pod is created? If yes, we will have to do validation too, otherwise we create situations where people can happily create VMIs but then wonder why it never starts because the controller can't create the corresponding pod.

Reply:
It looks like it could fail [here](https://github.com/kubernetes/kubernetes/blob/52d4500f238bb3b734e730027b2ca4c765a7a614/pkg/apis/core/validation/validation.go#L2632 ) and [here](https://github.com/kubernetes/kubernetes/blob/52d4500f238bb3b734e730027b2ca4c765a7a614/pkg/apis/core/validation/validation.go#L2673). Should I just duplicate these validation tests in `pkg/virt-api/webhooks/validating-webhook/validating-webhook.go`?

Reply:
yes please and also if you can add a comment on the validation functions that you add pointing the kubernetes validation.

Reply:
validating-webhook updated and full test coverage for the additions. I'll start work on the functional test next.